### DISCUSSION TITLE
How to represent constants in Web API

### DISCUSSION THREAD

**Discussion by skyclouds2001** [mdn/mdn-community#666] [Content]
some interfaces includes constants (both available on interface and its instance, works both like static property and instance property) to represent a special value with a specified meaning; but current way of how to represent is not consistent

for example, as for `XMLHTTPRequest` or `HTMLMediaElement`, its constants are not mentioned at all; for `Node`, its constants are mentioned under a property but not in a separate section; for `KeyboardEvent`, its constants has a separate *Constants* section to mention

An Opinionated solution is, create a separate *Constants* section to represent constants, like `KeyboardEvent` does

**Comment by teoli2003** [comment]
There are very few constants in the web platform interfaces.

This comes from the fact that "enumeration" (enum) are usually strings and can be extended over time.

Constants are, most of the time, just a special (old) case of enumerations. We should handle them in the same way we handle enums: in the parameters of the methods/properties that use/store them. I don't think we need a special section in the interface page, or individual pages.

**Reply by hamishwillee** [reply]
FYI this just came up in https://github.com/mdn/content/pull/33569, where constants were used to map enumerated values.

Completely agree this is an uncommon approach, and that the values should not get a special section. I just used them exactly as you have stated here - as the possible values of the parameter or possible values of data. That's more developer friendly than supplying a number.
